lib/index.js

```diff
@@ -838,24 +838,14 @@ class FontLibrary extends RustClass {
 }
 
 class Image extends RustClass {
-  #options
+  #fetch
+  #type // for future use by the SVG loader…
 
-  constructor(...args) {
+  constructor(width, height) {
```

Comment:
Well this is going to break stuff down the line and make it awkward to add future options. Like for SVG loading SVG from Buffer we need to indicate a type. Or in future, say you want to specify an element to render out of the SVG.  And the `raw` loader makes use of an options object as well.  I'm not sure adherence to standard does anything for us here.

Reply:
How would you check buffer content for SVG?  There's no magic byte header in a text SVG and it can start with various things (or nothing) before the actual <svg> tag shows up. Even a regex isn't trivial (and expensive). The opening `<xml>` tag isn't even actually required in a lot of cases.  🤷🏼 

I agree that needing to pass a special flag to indicate image type isn't the most elegant solution, but it's efficient and to the point, IMHO. I went back and forth on this a bit. Another thought was to add `svg` member to options, which for now can be a bool but later could be expanded to take an object with svg-specific options.  That's how we did it for `raw` since it has some options like color type to use (and height/width which wasn't part of Image at that point).

There's no browser API to load images from Buffers anyway. And I'm not seeing how having another way to init Image can't co-exist with the standard. Just being able to pass in an object which has height/width properties is nice syntax sugar if nothing else.

Reply:
I suspect this sort of data-format checking would work better on the rust side anyway (e.g., first try `Image::from_encoded` then (if it returns `None`) try `svg::Dom::from_bytes`, etc.).

Reply:
Really? Just to avoid the user specifying a type? I assume Skia has to (at a minimum) scan the first X bytes for every known image type before it rejects it. Assuming it's even that smart. Sounds like a slowdown for SVGs when already having bytes in a buffer should be faster.

Reply:
Totally, I'd definitely want to do some benchmarking before committing to the approach.